tests/e2e/attention/generate_e2e_attention_tests.py

```diff
@@ -0,0 +1,499 @@
+        f"  %{name}_dim2 = arith.constant {tensor_shape[2]} : i64\n"
+        f"  %{name}_element_type = hal.element_type<{element_type.value}> : i32\n"
+        f"  %{name}_seed = arith.constant {pseudorandom_generator_seed} : i32\n"
+        f"  %{name} = call @attention_test.generate_random_tensor(%device, %{name}_dim0, %{name}_dim1, %{name}_dim2, %{name}_element_type, %{name}_seed) : (!hal.device, i64, i64, i64, i32, i32) -> !hal.buffer_view\n"
+    )
+
+
+call_id = 0
+
+
+def generate_call(
+    function: MLIRFunction,
+    query_type: QueryElemTypeId,
+    key_type: KeyElemTypeId,
+    value_type: ValueElemTypeId,
+    shapes_scale: TestShapeAndScale,
+):
+    global call_id
+    func_name = f"{function.name}_{shapes_scale.batch}_{shapes_scale.m}_{shapes_scale.k1}_{shapes_scale.k2}_{shapes_scale.n}_{shapes_scale.k1}_{shapes_scale.scale}"
+    func_name = f"{func_name}_{call_id}"
+    call_id = call_id + 1
+
+    description = f"Attention shape (BATCHxMxK1xK2xN): {shapes_scale.batch}x{shapes_scale.m}x{shapes_scale.k1}x{shapes_scale.k2}x{shapes_scale.k1}x{shapes_scale.n}"
+    op = (
+        f"func.func @{func_name}() attributes {{\n"
+        f'  iree.reflection = {{description = "{description}"}}\n'
+        "} {\n"
+        "  %device_index = arith.constant 0 : index\n"
+        "  %device = hal.devices.get %device_index : !hal.device\n"
+    )
+
+    query_shape, key_shape, value_shape, result_shape = get_tensor_shapes(
+        shapes_scale,
+    )
+
+    op = op + generate_random_3d_tensor("query", query_shape, query_type)
+    op = op + generate_random_3d_tensor("key", key_shape, key_type)
+    op = op + generate_random_3d_tensor("value", value_shape, value_type)
+
+    global pseudorandom_generator_seed
+    pseudorandom_generator_seed = pseudorandom_generator_seed - 1
+    op = op + (
+        f"  %scale = arith.constant {shapes_scale.scale} : f32\n"
+        f"  %result = call @module.{function.name}(%query, %key, %value, %scale) : (!hal.buffer_view, !hal.buffer_view, !hal.buffer_view, f32) -> !hal.buffer_view\n"
+    )
+
+    op = op + (
+        f"  %batch = arith.constant {shapes_scale.batch} : i64 \n"
+        f"  %m = arith.constant {shapes_scale.m} : i64 \n"
+        f"  %k1 = arith.constant {shapes_scale.k1} : i64 \n"
+        f"  %k2 = arith.constant {shapes_scale.k2} : i64 \n"
+        f"  %n = arith.constant {shapes_scale.n} : i64 \n"
+        f"  %query1 = hal.tensor.import %query : !hal.buffer_view -> tensor<{shapes_scale.batch}x{shapes_scale.m}x{shapes_scale.k1}xf16> \n"
+        f"  %key1 = hal.tensor.import %key : !hal.buffer_view -> tensor<{shapes_scale.batch}x{shapes_scale.k2}x{shapes_scale.k1}xf16> \n"
+        f"  %value1 = hal.tensor.import %value : !hal.buffer_view -> tensor<{shapes_scale.batch}x{shapes_scale.k2}x{shapes_scale.n}xf16> \n"
+        f"  %result1 = hal.tensor.import %result : !hal.buffer_view -> tensor<{shapes_scale.batch}x{shapes_scale.m}x{shapes_scale.n}xf16> \n"
```

Comment:
It can be renamed `query`, `key` and `value`. Remove the `1` suffix. Also, rename `queryExt1` to `queryExtBufferView`.

Reply:
sg